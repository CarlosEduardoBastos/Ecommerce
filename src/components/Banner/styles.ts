import styled, { css } from 'styled-components';
import {Container as TextComponent} from '../TextComponent/styles'


export const Container = styled.section`
  ${({theme}) => css`
    width: 100%;
    height: 100vh;
    background-color: ${theme.colors.primaryColor};
  `}
`;

export const Content = styled.div`
  ${({theme}) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 16px;

    .swiper-slide {
      display: flex;
      justify-content: center;
      flex-wrap: wrap;
      background: #fff;
      background: transparent;

      div{
        margin: 30px 0;
      }

      ${TextComponent}{
        display: block;
        max-width: 500px;
        margin: 15px 0;
      }

    }
  `}
`;

export const Text = styled.section`
  ${({theme}) => css`
    margin-top: 120px;
  `}
`;
export const Image = styled.div`
  ${({theme}) => css`
    max-width: 250px;
    max-height: 500px;
    img{
      object-fit: cover;
      object-position: center;
      height: 100%;
    }
  `}
`;