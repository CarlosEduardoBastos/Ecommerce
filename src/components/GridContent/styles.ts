import styled from 'styled-components';

export const Container = styled.div`
  display: grid;
  grid-template-columns: repeat(4,minMax(0,1fr));
  gap: 88px 30px;
`;
