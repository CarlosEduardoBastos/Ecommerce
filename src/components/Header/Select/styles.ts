import styled, { css } from 'styled-components';

export const Container = styled.select`
  ${({theme}) => css`
    text-decoration: none;
    text-transform: uppercase;
    font-weight: 700;
    color: ${theme.colors.darkColor}; 
    border: 2px solid #000;
    border-radius: 30px;
    padding: 10px 20px 9px;
    transition: .3s linear,color .3s linear;
    appearance: none;
    -webkit-appearance: none;
    -moz-appearance: none;
    background-image: url('../assets/chevron-down-regular-24.png');
    background-repeat: no-repeat;
    background-position: right center;
    padding-right: 20px;
  `}
`;
