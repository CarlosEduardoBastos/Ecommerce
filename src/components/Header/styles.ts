import styled from 'styled-components';

export const Container = styled.header`
  width: 100%;
`;

export const Header = styled.div`
  padding: 30px 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const Nav = styled.div`
  display: flex;
  grid-column-gap: 8px;
`;

export const Logo = styled.div`

`;

export const Actions = styled.div`

`;
