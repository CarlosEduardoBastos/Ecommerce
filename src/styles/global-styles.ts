import { createGlobalStyle, css } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  ${({ theme }) => css`
    * {
      margin: 0;
      padding: 0;
      box-sizing: border-box;
    }
    body {
      
    }
    a {
      text-decoration: none;
    }
  `}
`;