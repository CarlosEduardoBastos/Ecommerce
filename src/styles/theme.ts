export const theme = {
  colors: {
    primaryColor: '#ffc700',
    secondaryColor: '#10a958',
    darkColor: '#212121',
    lightColor: '#f8f8f8'
  },
  fonts: {
    family: {
      default: "'Open Sans', sans-serif",
      secondaryFont: "'Oswald', sans-serif"
    },
    sizes: {
      small: '0.875rem',
      xsmall: '1.25rem',
      xxsmall: '1.375rem',
      medium: '1.5rem',
      xmedium: '1.625rem',
      large: '2rem',
      xlarge: '3rem',
    },
  },
  media: {
    lteMedium: '(max-width: 767px)',
  },
  opacity: {
  },
}as const;