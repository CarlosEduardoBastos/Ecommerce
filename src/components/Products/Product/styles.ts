import { Link } from 'react-router-dom';
import {Container as Heading} from '../../Heading/styles'
import styled, { css } from 'styled-components';

export const Container = styled(Link)`
  height: 463px;
  width: 310px;
  text-decoration: none;
`;

export const Image = styled.img`
  max-width: 100%;
  max-height: 100%;
`

export const ImageContainer = styled.div`
  border-radius: 35px;
  background-color: #dadad8;
  height: 412px;
`
export const Content = styled.div`
  max-width: 100%;
  display: flex;
  justify-content: space-between;
`

export const ContentText = styled.div`
  ${Heading}{
    font-weight: 700;
  }
`;

export const Price = styled.span`
  ${({theme}) => css`
    text-decoration: none;
    color: ${theme.colors.darkColor};
    font-size: ${theme.fonts.sizes.xxsmall};
    font-weight: 400;
  `}
`;

export const NewProduct = styled.span`
  ${({theme}) => css`
    color: ${theme.colors.secondaryColor};
    font-weight: 400;
    margin-bottom: 4px;
    font-size: ${theme.fonts.sizes.xxsmall};
  `}
`