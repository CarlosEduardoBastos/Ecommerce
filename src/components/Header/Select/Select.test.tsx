import { Select } from '.';
import { renderTheme } from '../../../styles/render-theme';

describe('<Select />', () => {
  it('should mach snapshot', () => {
    const { container } = renderTheme(<Select options={[
      { value: 'opcao1', label: 'Opção 1' },
      { value: 'opcao2', label: 'Opção 2' },
      { value: 'opcao3', label: 'Opção 3' },
    ]}/>);
    expect(container).toMatchSnapshot()
  });
});
