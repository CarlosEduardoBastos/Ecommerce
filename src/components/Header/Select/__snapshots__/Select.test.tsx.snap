// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should mach snapshot 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-weight: 700;
  color: #212121;
  border: 2px solid #000;
  border-radius: 30px;
  padding: 10px 20px 9px;
  -webkit-transition: .3s linear,color .3s linear;
  transition: .3s linear,color .3s linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  background-image: url('../assets/chevron-down-regular-24.png');
  background-repeat: no-repeat;
  background-position: right center;
  padding-right: 20px;
}

<div>
  <select
    class="c0"
  >
    <option
      value="opcao1"
    >
      Opção 1
    </option>
    <option
      value="opcao2"
    >
      Opção 2
    </option>
    <option
      value="opcao3"
    >
      Opção 3
    </option>
  </select>
</div>
`;
